File: spring/src/main/kotlin/com/windsnow1025/leximemo/spring/entity/Word.kt
package com.windsnow1025.leximemo.spring.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.*

@Entity
class Word(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int?,

    @Column(unique = true)
    var word: String,

    var translation: String,

    var exampleSentence: String,

    var frequency: Int,

    @OneToMany(mappedBy = "word")
    @JsonBackReference
    var userWords: MutableSet<UserWord> = mutableSetOf(),

    @ManyToMany(mappedBy = "words")
    @JsonManagedReference
    var dictionaries: MutableSet<Dictionary> = mutableSetOf()
)

File: next/src/model/Word.ts
import {UserWord} from "@/model/UserWord";
import {Dictionary} from "@/model/Dictionary";

export class Word {
  id?: number;
  word: string = '';
  translation: string = '';
  exampleSentence: string = '';
  frequency: number = 0;
  userWords: UserWord[] = [];
  dictionaries: Dictionary[] = [];

  constructor(id?: number, word: string = '', translation: string = '', exampleSentence: string = '', frequency: number = 0, userWords: UserWord[] = [], dictionaries: Dictionary[] = []) {
    this.id = id;
    this.word = word;
    this.translation = translation;
    this.exampleSentence = exampleSentence;
    this.frequency = frequency;
    this.userWords = userWords;
    this.dictionaries = dictionaries;
  }
}